ols/src/cache/in_memory_cache.py

```diff
@@ -4,8 +4,10 @@
 
 import threading
 from collections import deque
-from typing import Union
+from typing import TYPE_CHECKING, Union
 
+if TYPE_CHECKING:
+    from ols.app.models.config import MemoryConfig
```

Comment:
is this a runtime performance optimization or what?

Reply:
if I make that import unconditional, I get
```
ruff . --fix --per-file-ignores=tests/*:S101
ols/src/cache/in_memory_cache.py:10:35: TCH001 Move application import `ols.app.models.config.MemoryConfig` into a type-checking block
```

Reply:
weird.  @tisnik can you explain this?  It seems odd, especially that this seems to be the only place it's necessary to do this.

Reply:
it's runtime optimization indeed. The module is not used in the real code, "just" in type annotations. So in runtime it does not need to be imported.

Reply:
if `"MemoryConfig"` will be used instead (string), no conditional import will be needed